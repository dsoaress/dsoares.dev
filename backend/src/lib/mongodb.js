import { MongoClient } from 'mongodb'

const { MONGODB_URI, MONGODB_DB } = process.env

let cached = global.mongo

if (!cached) {
  cached = global.mongo = { conn: null, promise: null }
}

export async function connectToDatabase() {
  if (cached.conn) {
    return cached.conn
  }

  if (!cached.promise) {
    const opts = {
      useNewUrlParser: true,
      useUnifiedTopology: true
    }

    cached.promise = MongoClient.connect(MONGODB_URI, opts).then(client => {
      return {
        client,
        db: client.db(MONGODB_DB)
      }
    })
  }
  cached.conn = await cached.promise
  return cached.conn
}
