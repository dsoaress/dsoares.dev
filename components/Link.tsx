import NextLink from 'next/link'
import { lighten } from 'polished'
import { AnchorHTMLAttributes } from 'react'
import styled from 'styled-components'

const StyledLink = styled.a`
  transition: color 0.3s cubic-bezier(0.4, 0, 0.2, 1);
  color: ${({ theme }) => theme.colors.text};
  text-decoration: none;
  font-weight: 600;
  cursor: pointer;

  &:hover {
    color: ${({ theme }) => theme.colors.primary};
  }

  &:active {
    color: ${({ theme }) => lighten(0.1, theme.colors.primary)};
  }
`

export function Link({ href, ...props }: AnchorHTMLAttributes<HTMLAnchorElement>) {
  return (
    <NextLink href={href} passHref>
      <StyledLink {...props} />
    </NextLink>
  )
}
