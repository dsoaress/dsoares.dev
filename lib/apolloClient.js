import { ApolloClient, InMemoryCache } from '@apollo/client'

const { API_URL } = process.env

const client = new ApolloClient({
  uri: `${API_URL}/graphql`,
  cache: new InMemoryCache()
})

export default client
