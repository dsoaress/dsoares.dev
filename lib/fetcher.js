import { api } from './api'

export const fetcher = (...args) => api.get(...args).then(res => res.data)
