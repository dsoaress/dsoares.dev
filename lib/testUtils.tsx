import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

const Providers = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'queries'>): RenderResult =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
