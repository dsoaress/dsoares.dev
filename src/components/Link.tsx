import cn from 'classnames'
import NextLink from 'next/link'
import type { AnchorHTMLAttributes, ReactNode } from 'react'
import { Link as ReactScroll } from 'react-scroll'

type LinkProps = {
  active?: boolean
  href: string
  locale?: string
  external?: boolean
} & AnchorHTMLAttributes<HTMLAnchorElement>

type ScrollLinkProps = {
  to: string
  className?: string
  children: ReactNode
}

export function Link({ active = false, href, className, locale, external, ...props }: LinkProps) {
  return (
    <NextLink href={href} locale={locale}>
      <a
        className={cn(className, { 'text-primary-500 dark:text-primary-500': active })}
        rel={external ? 'noopener noreferrer' : undefined}
        target={external ? '_blank' : undefined}
        {...props}
      />
    </NextLink>
  )
}

export function ScrollLink(props: ScrollLinkProps) {
  // TODO fix this
  // @ts-ignore
  return <ReactScroll smooth={true} offset={-76} href="/" aria-hidden {...props} />
}

export function CardLink({ className, ...props }: LinkProps) {
  return <Link className={cn('absolute inset-0', className)} {...props} />
}
