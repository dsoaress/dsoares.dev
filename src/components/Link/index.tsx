import NextLink from 'next/link'
import { AnchorHTMLAttributes } from 'react'

import { StyledLink } from './styles'

export { CardLink, ScrollLink } from './styles'

export type LinkProps = {
  active?: boolean
  href: string
  locale?: string
} & AnchorHTMLAttributes<HTMLAnchorElement>

export function Link({ active = false, href, locale, ...rest }: LinkProps) {
  return (
    <NextLink href={href} locale={locale} passHref>
      <StyledLink active={active} {...rest} />
    </NextLink>
  )
}
