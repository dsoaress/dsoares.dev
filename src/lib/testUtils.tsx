import { render } from '@testing-library/react'
import { ComponentType, ReactElement, ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles/theme'

type ProvidersProps = {
  children?: ReactNode
}

const Providers = ({ children }: ProvidersProps) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
