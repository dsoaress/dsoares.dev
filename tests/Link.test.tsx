import Link from '@/components/Link'
import { render, screen } from '@/lib/testUtils'

describe('<Link />', () => {
  it('Renders with children', () => {
    render(<Link href="#">Children</Link>)

    expect(screen.getByText('Children')).toBeInTheDocument()
    expect(screen.getByText('Children')).toMatchSnapshot()
  })
})
