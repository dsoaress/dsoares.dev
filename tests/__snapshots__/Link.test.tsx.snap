// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Renders with children 1`] = `
.c0 {
  -webkit-transition: color 0.3s cubic-bezier(0.4,0,0.2,1);
  transition: color 0.3s cubic-bezier(0.4,0,0.2,1);
  color: #D1D5DB;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  cursor: pointer;
}

.c0:hover {
  color: #F43F5E;
}

.c0:active {
  color: #f76f86;
}

<a
  class="c0"
  href="/#"
>
  Children
</a>
`;
